servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -7750,6 +7750,172 @@ public void executeDeleteAssertionCondition(ResourceContext ctx, String domainNa
         }
     }
 
+    public void putDomainDependency(ResourceContext ctx, String domainName, String service, String auditRef, String caller) {
+
+        // our exception handling code does the check for retry count
+        // and throws the exception it had received when the retry
+        // count reaches 0
+
+        for (int retryCount = defaultRetryCount; ; retryCount--) {
+
+            try (ObjectStoreConnection con = store.getConnection(false, true)) {
+
+                final String principal = getPrincipalName(ctx);
+
+                // first verify that auditing requirements are met
+
+                checkDomainAuditEnabled(con, domainName, auditRef, caller, principal, AUDIT_TYPE_DOMAIN);
+
+                // verify domain exists
+
+                Domain domain = con.getDomain(domainName);
```

Comment:
@havetisyan  I've put the check if the domain exists in both putDomainDependency and deleteDomainDependency even though in the existing tests it the function checkDomainAuditEnabled (that is called just before this one) checks the domain.
The reason it is checked in tests is that we set the property ZMS_PROP_AUDIT_REF_CHECK_OBJECTS during zms initiation to all audit types. 
By default, however, we only set the "role" and "group" bits (in our internal environments, we seem to use the default value). In the dependency tests, I cleared this property and now the domain is checked even when the "ZMS_AUDIT_TYPE_DOMAIN" isn't set.